pkg/maps/maps.go

```diff
@@ -58,3 +58,14 @@ func MergeCopy[M1 ~map[K]V, M2 ~map[K]V, K comparable, V any](base M1, override
 	maps.Copy(dst, override)
 	return dst
 }
+
+// Match checks if all key-value pairs in 'subset' are present and have matching values in 'superset'.
+// It returns true only if every key in 'subset' exists in 'superset' and their corresponding values are equal.
+func Match[M1, M2 ~map[K]V, K comparable, V comparable](superset M1, subset M2) bool {
```

Comment:
to avoid naming confusion with the `Equal` method, should this one be called `SupersetOf`? Similar to what was done for our `sets` implementation: https://github.com/istio/istio/blob/master/pkg/util/sets/set.go#L156

Reply:
Yes, this name is more straightforward I think. I'll revise.

Reply:
seems reasonable